internal/handler/handler_test.go

```
@@ -0,0 +1,746 @@
+			method:   "GET",
+			expected: 200,
+		},
+		{
+			URL:      "http://localhost:8080/positions?limit=asd&offset=1",
+			method:   "GET",
+			expected: 500,
+		},
+		{
+			URL:      "http://localhost:8080/positions?limit=1&offset=asd",
+			method:   "GET",
+			expected: 500,
+		},
+		{
+			URL:      "http://localhost:8080/positions?limit=110&offset=3",
+			method:   "GET",
+			expected: 500,
+		},
+	}
+
+	for _, testCase := range testTable {
+		r, err := http.NewRequest(testCase.method, testCase.URL, nil)
+		if err != nil {
+			t.Fatalf("error: %v", err)
+		}
+		r = createTestContext(r)
+		w := httptest.NewRecorder()
+		handler.GetPositions(w, r)
+		result := w.Result()
+		defer result.Body.Close()
+		if result.StatusCode != testCase.expected {
+			t.Fatalf("expected %v; get %v", testCase.expected, result)
+		}
+	}
+}
+
+func TestHand_UpdateEmployee(t *testing.T) { //nolint: funlen
+	initTest()
+	firstPosition := internal.Position{ID: createPosID(), Salary: decimal.New(500, 0), Name: "worker"}
+	repos.AddPosition(&firstPosition)
+	id, err := uuid.Parse(positionIDs[0])
+	if err != nil {
+		t.Fatalf("Error: %v ", err)
+	}
+	firstEmployee := internal.Employee{ID: createEmpID(), FirstName: "Bob", LasName: "Vik", PositionID: id}
+	firstEmployeeID := firstEmployee.ID
+	repos.AddEmployee(&firstEmployee)
+	secondEmployee := internal.Employee{ID: firstEmployeeID, FirstName: "Victor", LasName: "Vik", PositionID: id}
+	thirdEmployee := internal.Employee{ID: createEmpID(), FirstName: "Fox", LasName: "Fok", PositionID: uuid.New()}
+	fourthEmployee := internal.Employee{ID: uuid.Nil, FirstName: "", LasName: "Fok", PositionID: uuid.New()}
+	fifthEmployee := internal.Employee{ID: firstEmployeeID, FirstName: "Chu", LasName: "Xi", PositionID: uuid.New()}
+	jsonSecondEmployee, _ := json.Marshal(secondEmployee)
+	jsonThirdEmployee, _ := json.Marshal(thirdEmployee)
+	jsonFourthEmployee, _ := json.Marshal(fourthEmployee)
+	jsonFifthEmployee, _ := json.Marshal(fifthEmployee)
```

Ignoring errors is a bad practice

---

+

---

It doesn't work.